skrub/_dataframe/_common.py

```diff
@@ -894,3 +927,20 @@ def _replace_regex_pandas(column, pattern, replacement):
 @replace_regex.specialize("polars")
 def _replace_regex_polars(column, pattern, replacement):
     return column.str.replace_all(pattern, replacement, literal=False)
+
+
+@dispatch
+def with_columns(df, col_name, values):
+    raise NotImplementedError()
+
+
+@with_columns.specialize("pandas", argument_type="DataFrame")
+def _with_columns_pandas(df, col_name, values):
+    df[col_name] = values
```

Comment:
here this modifies the input df in place. We should return a different dataframe instead and leave the input unchanged. you can use df.assign

Reply:
also do we want to allow passing a Series/Column rather than just values? That would depend on how this function is used but I don't see it being used anywhere? maybe if you don't need it here we can defer its addition until we actually need it

Reply:
also if it accepts just 1 column I guess it should be `with_column`? but then we might want to generalize it for several columns

Reply:
You're right, this implementation only works for adding one columns

As discussed below, I'm removing it from this PR since I only need in one place